internal/cli/cmd/plugin/build/build.go

```diff
@@ -81,22 +95,169 @@ func (o *option) Execute() error {
 		return fmt.Errorf("unable to read manifest: %w", err)
 	}
 
-	// Then we need to create the archive.
-	// The archive contains the following file:
+	// Check if we need to vendor the dependencies:
+	// - Parse the module file into a CUE instance
+	ctx := cuecontext.New()
+	insts := load.Instances([]string{moduleFile}, nil)
+	module := ctx.BuildInstance(insts[0])
+	if err := module.Err(); err != nil {
+		return fmt.Errorf("failed to load plugin module file: %w", err)
+	}
+
+	// - if no `deps` are present, skip vendoring
+	deps := module.LookupPath(cue.ParsePath("deps"))
+	if deps.Err() != nil {
+		logrus.Debugf("`deps` not found, no CUE dependencies to vendor")
+	} else {
+		// Vendor the dependencies
+		if err := o.vendorCueDependencies(); err != nil {
+			return fmt.Errorf("failed to vendor CUE dependencies: %w", err)
+		}
+		// restore the original state on exit
+		defer func() {
+			if err := os.RemoveAll(vendorDir); err != nil {
+				logrus.Printf("failed to cleanup %s: %v", vendorDir, err)
+			}
+		}()
```

Comment:
Technically a plugin could mix both old dependency management & new, which means that in this case this piece of code will NOT restore the original state. People would still be able to manually `git checkout --` afterwards though. IMHO this is a sufficiently acceptable impediment + corner case anyway to not go through the hassle of making the logic smarter here (would require to backup the whole cue.mod/pkg folder & restore it at the end). WDYT?

Reply:
perhaps we could simply check if the folder `cue.mode/pkg` already exists ?

A more complex but not horrible solution would be:
- verify with git there is no file that are not tracked
- run the build command
- remove the files that git returned as not tracked in `cue.mod/pkg`, so the build command doesn't seem to alter the system.

The second proposal wouldn't be that horrible from my opinion. What do you prefer ?

Reply:
Let's go for first option as agreed to not be opinionated on VCS usage